python/tank/pipelineconfig.py

```diff
@@ -559,54 +531,101 @@ def get_install_location(self):
         
         return core_api_root
 
-    def get_bundles_location(self):
+    def get_core_python_location(self):
         """
-        Returns the location where all apps/frameworks/engines are stored in subfolders
+        Returns the python root for this install.
         
         :returns: path string
         """
-        return os.path.join(self.get_install_location(), "install")
+        return os.path.join(self.get_install_location(), "install", "core", "python")
+
+
+    ########################################################################################
+    # accessing cached code
 
-    def get_apps_location(self):
+    def _preprocess_location(self, location_dict):
         """
-        Returns the location where apps are stored
-        
-        Note! This method has been deprecated and will be removed at 
-        some point in the future. Please do not use it.
-        
-        :returns: path string        
+        Preprocess location dict to resolve config-specific constants and directives.
+
+        This is only relevant if the locator system is used in conjunction with
+        a toolkit configuration. For example, the keyword {PIPELINE_CONFIG} is
+        only meaningful if used in the context of a configuration.
+
+        Location dictionaries defined and used outside of the scope of a
+        pipeline configuration do not support such keywords (since no
+        pipeline configuration exists at that point).
+
+        :param location_dict: Location dict to operate on
+        :param pipeline_config: Pipeline Config object
+        :returns: location dict with any directives resolved.
         """
-        return os.path.join(self.get_bundles_location(), "apps")
 
-    def get_engines_location(self):
+        if location_dict.get("type") == "dev":
+
+            # platform specific location support
+            platform_key = {"linux2": "linux_path", "darwin": "mac_path", "win32": "windows_path"}[sys.platform]
```

Comment:
@jfboismenu your comment doesn't feel directly related to this CR, so I'll skip that for now. Enough changes already as it is :) Maybe it's cleaner to introduce wrapper methods, but if i understand you correctly, we are not adding any logic of our own here, right? `is_linux()` is equal to `sys.platform == "linux2"`.